recipes/imagemagick/all/conanfile.py

```diff
@@ -0,0 +1,399 @@
+
+        suffix = {'MT': 'StaticMT',
+                  'MTd': 'StaticMTD',
+                  'MD': 'DynamicMT',
+                  'MDd': 'DynamicMT'}.get(str(self.settings.compiler.runtime))
+
+        # GdiPlus requires C++, but ImageMagick has *.c files
+        project = 'IM_MOD_emf_%s.vcxproj' % suffix if self.options.shared else 'CORE_coders_%s.vcxproj' % suffix
+        tools.replace_in_file(os.path.join('VisualMagick', 'coders', project),
+                              '<ClCompile Include="..\\..\\ImageMagick\\coders\\emf.c">',
+                              '<ClCompile Include="..\\..\\ImageMagick\\coders\\emf.c">\n'
+                              '<CompileAs>CompileAsCpp</CompileAs>')
+
+        for module in self._modules:
+            with tools.chdir(os.path.join('VisualMagick', module)):
+                msbuild = MSBuild(self)
+                msbuild.build(project_file='CORE_%s_%s.vcxproj' % (module, suffix),
+                              upgrade_project=False,
+                              platforms={'x86': 'Win32', 'x86_64': 'x64'})
+
+        with tools.chdir(os.path.join('VisualMagick', 'coders')):
+            pattern = 'IM_MOD_*_%s.vcxproj' % suffix if self.options.shared else 'CORE_coders_%s.vcxproj' % suffix
+            projects = glob.glob(pattern)
+            for project in projects:
+                msbuild = MSBuild(self)
+                msbuild.build(project_file=project,
+                              upgrade_project=False,
+                              platforms={'x86': 'Win32', 'x86_64': 'x64'})
+
+    def _build_configure(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=self._is_mingw_windows)
+
+        # workaround for xorg/system adding systme includes https://github.com/conan-io/conan-center-index/issues/6880
+        xft_path = os.path.join(self.build_folder, "xft.pc")
+        if os.path.exists(xft_path):
+            os.remove(xft_path)
+        if "/usr/include/uuid" in self._autotools.include_paths:
+            self._autotools.include_paths.remove("/usr/include/uuid")
+        if "/usr/include/freetype2" in self._autotools.include_paths:
+            self._autotools.include_paths.remove("/usr/include/freetype2")
+        for include in self._autotools.include_paths:
+            if include.startswith("/usr/include/libpng"):
+                self._autotools.include_paths.remove(include)
+
+        args = ['--disable-openmp',
+                '--disable-docs',
+                '--with-perl=no',
+                '--with-x=no'
+                ]
+
+        if self.options.shared:
+            args.extend(['--enable-shared=yes', '--enable-static=no'])
+        else:
+            args.extend(['--enable-shared=no', '--enable-static=yes'])
+        args.append('--enable-hdri=yes' if self.options.hdri else '--enable-hdri=no')
+        args.append('--with-quantum-depth=%s' % self.options.quantum_depth)
```

Comment:
To make this more readable imho, you can use the following pattern:
```python
yes_no = lambda v: "yes" if v else "no"
args = [
    "--enable-shared={}".format(yes_no(self.options.shared)),
    "--enable-static={}".format(yes_no(not self.options.shared)),
    "--enable-hdri={}".format(yes_no(self.options.hdri)),
    "--with-quantum-depth={}".format(self.options.quantum_depth),
    "--with-zlib={}".format(yes_no(self.options.with_zlib)),
...
]
```

Reply:
This appears to not follow PEP8, I've simplified it like so:
```py
        def yes_no(o):
            return "yes" if o else "no"

        args = [
            "--enable-shared={}".format(yes_no(self.options.shared)),
            "--enable-static={}".format(yes_no(not self.options.shared)),
            "--enable-hdri={}".format(yes_no(self.options.hdri)),
            "--with-quantum-depth={}".format(self.options.quantum_depth),
            "--with-zlib={}".format(yes_no(self.options.with_zlib)),
            "--with-bzlib={}".format(yes_no(self.options.with_bzlib)),
            "--with-lzma={}".format(yes_no(self.options.with_lzma)),
            "--with-lcms={}".format(yes_no(self.options.with_lcms)),
            "--with-openexr={}".format(yes_no(self.options.with_openexr)),
            "--with-jpeg={}".format(yes_no(self.options.with_jpeg)),
            "--with-openjp2={}".format(yes_no(self.options.with_openjp2)),
            "--with-pango={}".format(yes_no(self.options.with_pango)),
            "--with-png={}".format(yes_no(self.options.with_png)),
            "--with-tiff={}".format(yes_no(self.options.with_tiff)),
            "--with-webp={}".format(yes_no(self.options.with_webp)),
            "--with-xml={}".format(yes_no(self.options.with_xml2)),
            "--with-freetype={}".format(yes_no(self.options.with_freetype)),
            "--with-utilities={}".format(yes_no(self.options.utilities)),
            "--with-djvu={}".format(yes_no(self.options.with_djvu)),
        ]
```

Reply:
We're flexible in pep8 related things :smile: